src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostDocumentCompletionEndpointTest.cs

```diff
@@ -0,0 +1,508 @@
+             completionContext: new RoslynVSInternalCompletionContext()
+             {
+                 InvokeKind = RoslynVSInternalCompletionInvokeKind.Explicit,
+                 TriggerCharacter = null,
+                 TriggerKind = RoslynCompletionTriggerKind.Invoked
+             },
+             expectedItemLabels: ["char", "DateTime", "Exception"],
+             expectedItemCount: 1000);
+    }
+
+    [Fact]
+    public async Task CSharpClassMembersInCodeBlock()
+    {
+        await VerifyCompletionListAsync(
+            input: $$"""
+                This is a Razor document.
+
+                <div></div>
+
+                @code{
+                    void foo()
+                    {
+                        DateTime.$$
+                    }
+                }
+
+                The end.
+                """,
+             completionContext: new RoslynVSInternalCompletionContext()
+             {
+                 InvokeKind = RoslynVSInternalCompletionInvokeKind.Typing,
+                 TriggerCharacter = ".",
+                 TriggerKind = RoslynCompletionTriggerKind.TriggerCharacter
+             },
+             expectedItemLabels: ["DaysInMonth", "IsLeapYear", "Now"],
+             expectedItemCount: 20);
+    }
+
+    // Tests MarkupTransitionCompletionItemProvider
+    [Fact]
+    public async Task CSharpMarkupTransitionAndTagHelpersInCodeBlock()
+    {
+        await VerifyCompletionListAsync(
+            input: $$"""
+                This is a Razor document.
+
+                <div></div>
+
+                @code{
+                    void foo()
+                    {
+                        <$$
+                    }
+                }
+
+                The end.
+                """,
+             completionContext: new RoslynVSInternalCompletionContext()
+             {
+                 InvokeKind = RoslynVSInternalCompletionInvokeKind.Typing,
+                 TriggerCharacter = "<",
+                 TriggerKind = RoslynCompletionTriggerKind.TriggerCharacter
+             },
+             expectedItemLabels: ["text", "EditForm", "InputDate"],
+             expectedItemCount: 34);
+    }
+
+    [Fact]
+    public async Task RazorDirectives()
+    {
+        await VerifyCompletionListAsync(
+            input: $$"""
+                @$$
+                This is a Razor document.
+
+                <div></div>
+
+                @code{
+                    void foo()
+                    {
+                        
+                    }
+                }
+
+                The end.
+                """,
+             completionContext: new RoslynVSInternalCompletionContext()
+             {
+                 InvokeKind = RoslynVSInternalCompletionInvokeKind.Typing,
+                 TriggerCharacter = "@",
+                 TriggerKind = RoslynCompletionTriggerKind.TriggerCharacter
+             },
+             expectedItemLabels: ["using", "using directive ...", "page", "page directive ..."],
```

Comment:
Should this include a couple of C# completion items too?

Reply:
Added both suggestions.